assets/js/payplug-integrated-payments.js

```diff
@@ -1,122 +1,262 @@
+					IntegratedPayment.props.return_url = result.data.return_url;
+					IntegratedPayment.SubmitPayment();
 				} else {
-					Integrated.save_card = false;
-					$saveCard.removeClass('-checked');
+					alert("NOT CREATED")
 				}
+			},
+		});
+
+		function getFormData($form){
+			var unindexed_array = $form.serializeArray();
+			var indexed_array = {};
+
+			jQuery.map(unindexed_array, function(n, i){
+				indexed_array[n['name']] = n['value'];
 			});
+
+			return indexed_array;
+		}
+	},
+	SubmitPayment: function(){
+		try {
+			IntegratedPayment.props.api.pay(IntegratedPayment.props.paymentId, Payplug.Scheme.AUTO, {save_card: false});
+		} catch(error) {
+			console.log(error);
 		}
 	}
+};
+
+jQuery( 'body' ).on( 'updated_checkout', function() {
+	IntegratedPayment.init();
 
-	PayplugIntegrated = {};
-
-	//TODO:: remove this method once development is done
-	PayplugIntegrated.showErrors = function(){
-		elems = document.querySelectorAll(".IntegratedPayment_error span");
-		elems.forEach(function (elem, index) {
-			if(elem.classList.contains('-hide')){
-				elem.classList.remove('-hide');
-				elem.classList.add("-show");
-			}else{
-				elem.classList.remove('-show');
-				elem.classList.add("-hide");
+	IntegratedPayment.props.pan.onChange(function(err) {
+		if (err.error) {
+			document.querySelector(".payplug.IntegratedPayment_error.-pan").classList.remove("-hide");
+			document.querySelector('.pan-input-container').classList.add("-invalid");
+
+			if (err.error.name === "FIELD_EMPTY") {
+				document.querySelector(".payplug.IntegratedPayment_error.-pan").querySelector(".emptyField").classList.remove("-hide");
+				document.querySelector(".payplug.IntegratedPayment_error.-pan").querySelector(".invalidField").classList.add("-hide");
+			} else {
+				document.querySelector(".payplug.IntegratedPayment_error.-pan").querySelector(".invalidField").classList.remove("-hide");
+				document.querySelector(".payplug.IntegratedPayment_error.-pan").querySelector(".emptyField").classList.add("-hide");
 			}
-		});
-		elems = document.querySelectorAll(".IntegratedPayment_error");
-		elems.forEach(function (elem, index) {
-			if(elem.classList.contains('-hide')){
-				elem.classList.remove('-hide');
-				elem.classList.add("-show");
-			}else{
-				elem.classList.remove('-show');
-				elem.classList.add("-hide");
+		} else {
+			document.querySelector(".payplug.IntegratedPayment_error.-pan").classList.add("-hide");
+			document.querySelector('.pan-input-container').classList.remove("-invalid");
+			document.querySelector(".payplug.IntegratedPayment_error.-pan").querySelector(".invalidField").classList.add("-hide");
+			document.querySelector(".payplug.IntegratedPayment_error.-pan").querySelector(".emptyField").classList.add("-hide");
+			IntegratedPayment.props.fieldsValid.pan = true;
+			IntegratedPayment.props.fieldsEmpty.pan = false;
+		}
+	});
+
+	IntegratedPayment.props.cvv.onChange(function(err) {
```

Comment:
Can you create a generic method to handle the errors.

`
field = '.-exp';
document.querySelector(".payplug.IntegratedPayment_error"+field)
`

ofc for the all on change, and you would only send the `field` as a parameter, from what I'm seing every code is the same for the validation on each field